src/main/java/frc/robot/subsystems/swerve/ModuleIOReal.java

```diff
@@ -224,24 +232,6 @@ public void updateInputs(final ModuleIOInputs inputs, final List<Samples> asyncO
     inputs.turnVelocityRadPerSec = Units.rotationsToRadians(turnVelocity.getValueAsDouble());
     inputs.turnAppliedVolts = turnAppliedVolts.getValueAsDouble();
     inputs.turnCurrentAmps = new double[] {turnCurrent.getValueAsDouble()};
-
-    inputs.odometryTimestamps =
-        asyncOdometrySamples.stream().mapToDouble(s -> s.timestamp()).toArray();
-    inputs.odometryDrivePositionsMeters =
-        asyncOdometrySamples.stream()
-            .map(s -> s.values().get(drivePosition))
-            .map(NullableDouble::new)
-            .toArray(NullableDouble[]::new);
-    inputs.odometryTurnPositions =
-        asyncOdometrySamples.stream()
-            // should be after offset + gear ratio
-            .map(s -> s.values().get(turnPosition))
-            .map(
-                d ->
-                    d == null
-                        ? new NullableRotation2d(null)
-                        : new NullableRotation2d(Rotation2d.fromRotations(d)))
```

Comment:
Should we just cut the nullable wrappers? Is the assumption that we just won't see the result vs supporting an explicit null result? You explicitly don't handle the case where we have fewer than 4 modules returning valid data - have we tested the case where that happens or where the gyro gets disconnected?

Reply:
have not tested that yet, which is important. will try to get to it on sunday. in that case I believe `updateOdometry` should see a null module position or gyro angle and handle it.